gms/gossiper.cc

```diff
@@ -946,6 +941,7 @@ future<> gossiper::replicate_live_endpoints_on_change() {
 
         if (live_endpoint_changed) {
             _shadow_live_endpoints = _live_endpoints;
+            ++_live_endpoints_version;
```

Comment:
> Just an idea, not sure if worth pursuing.

I'm trying to keep the scope of this series small.
There are indeed many ideas for improvement.
Let's consider that as a followup...

My main goal is to proceed to the next step which to make the changes to the endpoint state values atomic  and exception safe by preparing the modified endpoint_state maps in temporary variables instead of in place as it is done in many places today, and then let `replicate()` apply them in an exception safe way on all shards or on none.

Then the next step is extend the endpoint states model to support dual-indexing, by inet_address and by host_id and providing a set of entry point to get to the endpoint state by host_id, and converge all users to use them, and eventually drop the address-based entry points we won't use.